File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/VigoSourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class VigoSourceApplication extends BaseSourceApplication {
    public VigoSourceApplication(@Value("${fint.flyt.vigo.sso.client-id:#{null}}") String clientId) {
        super(4L, "VIGO", clientId);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/VigoSourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class VigoSourceApplication(
    @Value("\${fint.flyt.vigo.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.vigo.available:true}")
    available: Boolean,
) : BaseSourceApplication(
        id = 4L,
        displayName = "VIGO",
        clientId = clientId,
        available = available,
    )
